mpas_analysis/test/test_timekeeping.py

```diff
@@ -0,0 +1,200 @@
+"""
+Unit test infrastructure for the Date class
+
+Xylar Asay-Davis
+11/02/2016
+"""
+
+import pytest
+import datetime
+from mpas_analysis.shared.timekeeping.MpasRelativeDelta \
+    import MpasRelativeDelta
+from mpas_analysis.test import TestCase
+from mpas_analysis.shared.timekeeping.utility import stringToDatetime, \
+    stringToRelativedelta, clampToNumpyDatetime64
+
+
+class TestTimekeeping(TestCase):
+    def test_timekeeping(self):
+
+        # test each possible format:
+        # YYYY-MM-DD_hh:mm:ss
+        # YYYY-MM-DD_hh.mm.ss
+        # YYYY-MM-DD_SSSSS
+        # DDD_hh:mm:ss
+        # DDD_hh.mm.ss
+        # DDD_SSSSS
+        # hh.mm.ss
+        # hh:mm:ss
+        # YYYY-MM-DD
+        # SSSSS
+
+        # test datetime.datetime
+        # YYYY-MM-DD_hh:mm:ss
+        date1 = stringToDatetime('0001-01-01_00:00:00')
+        date2 = datetime.datetime(year=1, month=1, day=1, hour=0, minute=0,
+                                  second=0)
+        self.assertEqual(date1, date2)
+
+        # test relativedelta
+        # YYYY-MM-DD_hh:mm:ss
+        # gregorian_noleap
+        calendar = 'gregorian_noleap'
+        date1 = stringToRelativedelta('0001-00-00_00:00:00',
+                                      calendar=calendar)
+        date2 = MpasRelativeDelta(years=1, months=0, days=0, hours=0,
+                                  minutes=0, seconds=0, calendar=calendar)
+        self.assertEqual(date1, date2)
+
+        # gregorian
+        calendar = 'gregorian'
```

Comment:
I guess I don't really understand what you mean by "test the whole range of acceptable dates"?  I guess you mean the different string formattings?  I think it's pretty redundant to check this for both calendars because the function for parsing these strings is completely independent of the calendar, it just keeps the calendar around for doing math operations later on.  So in any cases where we're not doing math, the calendar is irrelevant.